static/src/javascripts/projects/common/modules/identity/account-profile.js

```diff
@@ -96,8 +104,58 @@ define([
         }
     };
 
+    accountProfile.prototype.avatarUploadByGAE = function (avatarForm) {
+        var self = this;
+        var xhr = self.createCORSRequest('GET', self.urls.avatarTokenUrl);
+
+        if (!xhr) {
+            self.prependErrorMessage(self.messages.noCorsError, avatarForm);
+        }
+
+        xhr.onerror = function () {
+            self.prependErrorMessage(self.messages.noServerError, avatarForm);
+        };
+
+        xhr.onload = function () {
+            avatarForm.setAttribute('action', xhr.responseText);
+            avatarForm.submit();
+        };
+
+        xhr.send();
+    };
+
+    accountProfile.prototype.avatarUploadByApi = function (avatarForm) {
+        var self = this;
```

Comment:
To be honest, I was just trying to be consistent with the rest of the code here as I'm no javascript expert. How would I use bind instead?